subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -799,6 +793,20 @@ In this way, you can use `gradle test` to test JUnit 3/4 tests on JUnit Platform
 
 A sample of mixed tests can be found at `samples/testing/junitplatform/engine` in the '-all' distribution of Gradle.
 
+[[custom_test_engine]]
+===== Custom test engine
+
+With JUnit Platform, you can customize `TestEngine` implementation yourself. By default, `TestEngine` registration is done via Java's `java.util.ServiceLoader` mechanism.
+In most cases, all you need to do is registering your engine implementation as documented https://junit.org/junit5/docs/current/user-guide/#launcher-api-engines-custom[here].
```

Comment:
Not in most cases, it is always done via a ServiceLoader. The include/exclude filter simply allows you to filter the set of engines that will be used. Instead we could shorten this to:

By default, all engines on the test runtime classpath will be used. If you want to control the engines used explicitly, you can do as follows: